static/app/components/assigneeSelector.tsx

```diff
@@ -296,17 +296,41 @@ class AssigneeSelector extends React.Component<Props, State> {
   renderNewDropdownItems(): ItemsBeforeFilter {
     const teams = this.renderNewTeamNodes();
     const members = this.renderNewMemberNodes();
+    const suggestedAssignees = this.renderSuggestedAssigneeNodes() ?? [];
+    const assigneeIds = new Set();
+    const filteredTeams: ItemsBeforeFilter = [];
+    const filteredMembers: ItemsBeforeFilter = [];
+
+    // filter out duplicates of Team/Member if also a Suggested Assignee
+    if (suggestedAssignees.length) {
+      suggestedAssignees.filter(assignee =>
+        assigneeIds.add(`${assignee.value.type}:${assignee.value.assignee.id}`)
+      );
+      teams.filter(team => {
+        return assigneeIds.has(`${team.value.type}:${team.value.assignee.id}`)
+          ? null
+          : filteredTeams.push(team);
+      });
+      members.filter(member => {
+        return assigneeIds.has(`${member.value.type}:${member.value.assignee.id}`)
+          ? null
+          : filteredMembers.push(member);
+      });
+    }
 
     const dropdownItems: ItemsBeforeFilter = [
-      {label: this.renderDropdownGroupLabel(t('Teams')), id: 'team-header', items: teams},
+      {
+        label: this.renderDropdownGroupLabel(t('Teams')),
+        id: 'team-header',
+        items: filteredTeams.length ? filteredTeams : teams,
```

Comment:
not sure if you want to fallback on `teams` if `filteredTeams` is empty. There is a case where all the teams get filtered out. For example say an org has one team only and that one team is a suggested assignee. Then all the teams would get filtered out in the previous code, resulting in an empty `filteredTeams` array. This is expected so we shouldn't fall back to showing all teams or else they would not get deduped. Same applies to the `filteredMembers`